commands.go

```diff
@@ -158,11 +238,25 @@ func (cmd Command) Run(label string, cmdline ...string) error {
 		defer services.Allow()
 	}
 
-	err := exe.Run()
+	// Save the original resolv.conf and copy version from host
+	resolvsum, err := cmd.saveResolvConf()
+	if err != nil {
+		return err
+	}
+
+	if err = exe.Run(); err != nil {
+		return err
```

Comment:
Previously we'd run flush and cleanup in case Run errored out; Now it bails immediately? Was that change intentional ?

Reply:
Nice catch! added defer for both